js/libs/keycloak-admin-client/src/defs/userProfileConfig.ts

```diff
@@ -10,7 +10,7 @@ export interface UserProfileAttribute {
   validations?: Record<string, unknown>;
   validators?: Record<string, unknown>;
   annotations?: Record<string, unknown>;
-  required?: UserProfileAttributeRequired;
+  required?: boolean | UserProfileAttributeRequired;
```

Comment:
@pedroigor could tell me whether the `UserProfileAttributeRequired` type is ever returned anywhere in the REST API? I get the feeling we made a mistake here, and this is actually always a boolean.

Reply:
You're right, it is a boolean.

It is probably there because the interface was generated from a user profile configuration. The type exists on the server side to represent the configuration for mandatory/optional attributes and it is not exposed when querying the metadata. What you get is a calculated value based (true|false) at the end.

Reply:
Peered up with Pedro to fix the types here, so we are good to go on this front.